CHANGELOG.next.toml

```diff
@@ -132,3 +130,167 @@ Fix request canonicalization for HTTP requests with repeated headers (for exampl
 references = ["smithy-rs#2261", "aws-sdk-rust#720"]
 meta = { "breaking" = false, "tada" = false, "bug" = true }
 author = "nipunn1313"
+
+[[aws-sdk-rust]]
+message = """Request IDs can now be easily retrieved on successful responses. For example, with S3:
+```rust
+// Import the trait to get the `request_id` method on outputs
+use aws_sdk_s3::types::RequestId;
+let output = client.list_buckets().send().await?;
+println!("Request ID: {:?}", output.request_id());
+```
+"""
+references = ["smithy-rs#76", "smithy-rs#2129"]
+meta = { "breaking" = true, "tada" = false, "bug" = false }
+author = "jdisanti"
+
+[[aws-sdk-rust]]
+message = """Retrieving a request ID from errors now requires importing the `RequestId` trait. For example, with S3:
+```rust
+use aws_sdk_s3::types::RequestId;
+println!("Request ID: {:?}", error.request_id());
+```
+"""
+references = ["smithy-rs#76", "smithy-rs#2129"]
+meta = { "breaking" = true, "tada" = false, "bug" = false }
+author = "jdisanti"
+
+[[smithy-rs]]
+message = "Generic clients no longer expose a `request_id()` function on errors. To get request ID functionality, use the SDK code generator."
```

Comment:
This seems too harsh. Whitelabel clients should also be able to easily opt into retrieving request IDs without having to onboard onto the full SDK code generator --- perhaps the decorator could be enabled via a codegen config key?

Reply:
If I understand correctly, a whitelabel client would already be using other parts of sdk-codegen for things like credentials, right? So those would be able to also use request ID after these changes. If there is a use case for pure/generic Smithy clients (generated by `codegen-client`) to have request ID, then it really should be in the Smithy spec.

Reply:
I use the terms whitelabel and generic interchangeably, to refer to clients produced by `codegen-client`, without the SDK decorators in `sdk-codegen`.

There are use cases for generic clients to retrieve request IDs from responses, services return unmodeled request IDs frequently.

Reply:
I think we should add these kinds of features to `sdk-codegen`, but make them consumable in isolation via the `smithy-build.json` configuration. Essentially we want varying degrees of the AWS SDK.